### DISCUSSION TITLE
Get Total Number of Documents in SFTP Folder

### DISCUSSION THREAD

**Discussion by JJWest12** [nextgenhealthcare/connect#5623] [Q&A] [answered]
I have a file reader channel that connects to an SFTP connection. Is there any way to get the total number of documents in directory that I am connecting to?

**Comment by pacmano1** [comment]
You will have to call jsch manually for that.

**Reply by pacmano1** [reply]
https://forums.mirthproject.io/forum/mirth-connect/support/15586-sftp-connection-within-javascript?p=86929#post86929

**Reply by JJWest12** [reply]
Hello! Thank you for the additional info! I actually found that looking to see if the 'pollComplete' source map was established (indicating we are at the last file in the sequence) and then capturing the 'pollSequenceId' sourceMap for that file solved my issue!

**Reply by pacmano1** [reply]
Oh sure - that works!   I thought you wanted to know _before_ you actually got the messages.

**Comment by pacmano1** [comment, marked as answer]
OP answered his own post.
